pyfar/dsp/levels.py

```diff
@@ -0,0 +1,63 @@
+"""
+Functions used to calculate the levels of signals.
+"""
+
+import numpy as np
+import scipy.signal as sps
+
+
+def time_weighted_levels(signal, weighting: str,
+                         reference=1) -> np.ndarray:
+    """
+    Calculates sound pressure levels with exponential time weighting
+    according to IEC / DIN EN 61672-1.
+
+    Paramters:
+    ---------
+
+    signal: Signal
+        The signal object to calculate the levels for
+
+    type: str
+        The time weighting type. Options are "F" (="fast") and "S" (="slow"),
+        which correspond to level decays of -34.7 dB and -4.3 dB,
+        respectively.
+
+    reference: float
+        The reference sound pressure in Pascal to calculate the level
+        for. Defaults to 1, in which case the returned level is in dBFS
+        (decibels to full scale). If the input signal is in Pascal, use
+        `2e-5` to obtain a sound pressure level (SPL).
+
+    Returns
+    -------
+    A numpy array of the same shape as `signal.time` with the current time
```

Comment:
I would suggest to return a TimeData object. This way the level can easily be plotted using `pf.plot.time(level, dB=False)`